### PR TITLE
Optimize apk

### PR DESCRIPTION
**Pull request by jfoucry**
Change `minifyEnabled` to `true`
Change `minimunSdkCompile`
Upgrade gradle version

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -7,15 +7,15 @@ android {
 
     defaultConfig {
         minSdkVersion 17
-        targetSdkVersion 30
+       targetSdkVersion 31
```
Comment (pnu-s) [inline app/build.gradle:10]:
```suggestion
        targetSdkVersion 31
```

**suggestion**: Correcting wrong indentation

**Review by pnu-s — CHANGES_REQUESTED** [review]
Why did the `targetSdkVersion` change?

I'm not sure whether this is related, but the check is red :boom:

File: app/build.gradle (hunk 1 above)
Comment (jfoucry) [reply app/build.gradle]: The `targetSdkVersion` change was suggested by `AndroidStudio`, may be because I update it to the lastest version.

**Comment by Jean-BaptisteC** [comment]
Add android:exported="false" into https://github.com/Exodus-Privacy/exodus-android-app/blob/236e0518037e5bf03a8eb0b80c565433b3b8b364/app/src/main/AndroidManifest.xml#L20

File: app/build.gradle (hunk 1 above)
Comment (pnu-s) [reply app/build.gradle]: @jfoucry I would advise to revert this change then, and see whether that works

**Comment by Jean-BaptisteC** [comment]
You can change this in next release (2.2.1)
I prefere if we don't have difference between Fdroid and Google Play

File: app/build.gradle (hunk 1 above)
Comment (jfoucry) [reply app/build.gradle]: ok.

**Comment by jfoucry** [comment]
@Jean-BaptisteC 
I don't understand what I'am suppose to do. Did you want me to revert  `android:exported="false"` in `AndroidManifest.xml` file?

**Comment by Jean-BaptisteC** [comment]
Ok for revert
https://developer.android.com/guide/topics/manifest/activity-element looks informations about android:exported

**Comment by Jean-BaptisteC** [comment]
Two solutions
- Merge for the next release (2.2.1)
  - Ok for four first commit (just change android:exported="true" to allow activities execution
  - minifyEnabled = "true" has no difference on size on apk

- Wait release #133
  - Bump Gradle
  - Bump Target SDK Version to 31

**Comment by jfoucry** [comment]
I did see what's still wrong. I close the PR.

**Comment by pnu-s** [comment]
Re-opened as it looks good to me.
I think it can go to the next release

@Jean-BaptisteC any opinion on this?

**Review by Jean-BaptisteC — APPROVED** [review]
OK for me
In PR for the next release, I add bump target sdk version to 31

**Pull request merged**